### PR TITLE
[NUI][TCSACR-487] DragAndDrop: Add drag and drop APIs

### PR DESCRIPTION
**Pull request by taehyub**
### Description of Change ###

Add drag and drop APIs

### API Changes ###

https://code.sec.samsung.net/jira/browse/TCSACR-487
- public static DragAndDrop Instance
- public void StartDragAndDrop(View sourceView, View shadowView, DragData dragData,  SourceEventHandler callback)
- public void AddListener(View targetView, DragAndDropEventHandler callback)
- public void RemoveListener(View targetView, DragAndDropEventHandler callback)

### CODE REVIEW

**Review by jaehyun0cho — COMMENTED** [review]
I think <since_tizen> tag is missing.

**Comment by taehyub** [comment]
> I think <since_tizen> tag is missing.

updated

File: src/Tizen.NUI/src/public/DragAndDrop/DragAndDrop.cs [hunk 1]
```diff
@@ -26,11 +26,10 @@ namespace Tizen.NUI
     /// <summary>
     /// DragAndDrop controls the drag objet and data.
     /// </summary>
+    /// <since_tizen> 10 </since_tizen>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2000: Dispose objects before losing scope", Justification = "It does not have ownership.")]
-    [EditorBrowsable(EditorBrowsableState.Never)]
     public class DragAndDrop : BaseHandle
     {
-        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/DragAndDrop/DragAndDrop.cs:33]: Just my curiosity :
Shouldn't we add since_tizen tag for these public handlers?

Comment (jaehyun0cho) [reply src/Tizen.NUI/src/public/DragAndDrop/DragAndDrop.cs:33]: You are right. public delegate also requires <since_tizen>. Please check the follows.
```
        /// <summary>                                                            
        /// Type of callback to render the frame with OpenGLES APIs.             
        /// If the return value of this callback is not 0, the eglSwapBuffers() will be called.
        /// </summary>                                                           
        /// <returns>The return value is not 0, the eglSwapBuffers() will be called.</returns>
        /// <since_tizen> 10 </since_tizen>                                      
        public delegate int GLRenderFrameDelegate(); 
```

Comment (taehyub) [reply src/Tizen.NUI/src/public/DragAndDrop/DragAndDrop.cs:33]: done

File: src/Tizen.NUI/src/public/DragAndDrop/DragEvent.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ namespace Tizen.NUI
     /// <summary>
     /// Source event type.
     /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
+    /// <since_tizen> 10 </since_tizen>
     public enum SourceEventType
```
[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.NUI/src/public/DragAndDrop/DragEvent.cs:27]: We do not recommend nested enum. add prefix `Drag`

**Comment by dongsug-song** [comment]
@taehyub 
Hello, please check the reviews above.
please resolve the conflicts also.
thank you.

**Comment by taehyub** [comment]
> @taehyub Hello, please check the reviews above. please resolve the conflicts also. thank you.

conflict resolved

**Review by dongsug-song — APPROVED** [review]
acr accepted, approved

**Pull request merged**